backend/src/src-bdpimport/src/main/java/com/bosch/osmi/sw360/bdp/datasink/thrift/ThriftExchange.java

```diff
@@ -0,0 +1,257 @@
+        List<License> licenses = null;
+        try {
+            licenses = thriftApi.getLicenseClient().getLicenses();
+        } catch (TException e) {
+            logger.error("Could not fetch License list:" + e);
+        }
+        return licenses;
+    }
+
+    private com.siemens.sw360.datahandler.thrift.users.UserService.Iface getUserClient() {
+        return thriftApi.getUserClient();
+    }
+
+    /**
+     * Add the Vendor to DB. Required fields are: fullname, shortname, url.
+     *
+     * @param vendor Vendor to be added
+     * @return VendorId-String from DB.
+     */
+    public String addVendor(Vendor vendor) {
+        String vendorId = null;
+        try {
+            vendorId = getVendorClient().addVendor(vendor);
+        } catch (TException e) {
+            logger.error("Could not add Vendor:" + e);
+        }
+        return vendorId;
+    }
+
+    private com.siemens.sw360.datahandler.thrift.vendors.VendorService.Iface getVendorClient() {
+        return thriftApi.getVendorClient();
+    }
+
+    /**
+     * Add the Component to DB. Required fields are: name.
+     *
+     * @param component Component to be added
+     * @param user
+     * @return ComponentId-String from DB.
+     */
+    public String addComponent(Component component, User user) {
+        String componentId = null;
+        try {
+            componentId = getComponentClient().addComponent(component, user);
+        } catch (TException e) {
+            logger.error("Could not add Component for user with email=[" + user.getEmail() + "]:" + e);
+        }
+        return componentId;
+    }
+
+    /**
+     * Add the Release to DB. Required fields are: name, version, componentId.
+     *
+     * @param release Release to be added
+     * @param user
+     * @return releaseId-String from DB.
+     */
+    public String addRelease(Release release, User user) {
+        String releaseId = null;
+        try {
+            releaseId = getComponentClient().addRelease(release, user);
+        } catch (TException e) {
+            logger.error("Could not add Release for user with email=[" + user.getEmail() + "]:" + e);
+        }
+        return releaseId;
+    }
+
+    public String addProject(Project project, User user) {
+        String projectId = null;
+        try {
+            projectId = thriftApi.getProjectClient().addProject(project, user);
+        } catch (TException e) {
+            logger.error("Could not add Project for user with email=[" + user.getEmail() + "]:" + e);
+        }
+        return projectId;
+    }
+
+    public String addLicense(License license, User user) {
+        List<License> licenses = null;
+        try {
+            licenses = thriftApi.getLicenseClient().addLicenses(Arrays.asList(license));
+        } catch (TException e) {
+            logger.error("Could not add License for user with email=[" + user.getEmail() + "]:" + e);
+        }
+        return Objects.isNull(licenses) ? null : licenses.get(0).getId();
```

Comment:
Weird thing, isn't it, that license service doesn't have a method `addLicense` and you have to resort to this trickery? To add a license, one's supposed to use `updateLicense`. And then one is left to wonder why `addLicenses` does not perform any permission checks.
So, in case of failure you can't honestly say that you couldn't add License for `user`, because you actually tried to do that anonymously.
I was going to suggest that you use `updateLicense`, but then noticed that you won't get the license or its id back. Obviously, the license service has some potential for improvement. Sigh.